test/lib/hakkasan.test.ts

```diff
@@ -0,0 +1,41 @@
+import { generateMockFetch } from 'test/testUtils';
+import {
+  parseHakkasanData,
+  fetchHakkasanByRef,
+  VALID_HAKKSAN_REFS,
+} from 'lib/hakkasan';
+import events from 'data/events.json';
+
+describe('parseHakkasanData(text)', () => {
+  test('should parse the argument of retrieveJSONP from Hakkasan JSON data', () => {
+    const data = { message: 'Test data' };
+    const dataText = JSON.stringify(data);
+    const text = `retrieveJSONP(${dataText})`;
+    const result = parseHakkasanData(text);
+    expect(result).toEqual(data);
+  });
+
+  test('should catch and rethrow errors', () => {
```

Comment:
A couple things here!
1. The test assertion seems like it's only testing the error case in which no argument is passed into the function, which wouldn't actually be likely due to the fact that we're using TypeScript.
2. I'd say we can improve this test to be more specific in which _types_ of specific errors can be caught within the function with varying different types of text.
3. I'd recommend against creating tests as simple/generic as this as it doesn't exactly test much and give any hints as to what errors would (or should) be expected depending on the input.

Reply:
Good point! Updated the test with a more realistic input (string input without a `retrieveJSONP()` invocation).